lib/ecto/filters.ex

```diff
@@ -102,28 +93,19 @@ defmodule Ecto.Filters do
 
       defp get_filter_params(params) do
         cond do
-          is_list(params) and Keyword.has_key?(params, @key) -> Keyword.get(params, @key)
-          is_map(params) and Map.has_key?(params, @key) -> Map.get(params, @key)
-          is_map(params) and Map.has_key?(params, Atom.to_string(@key)) -> Map.get(params, Atom.to_string(@key))
-          true -> params
-        end
-      end
+          is_list(params) and Keyword.has_key?(params, @key) ->
+            Keyword.get(params, @key)
 
-      defp defaults({key, value}, query) do
-        source = elem(query.from.source, 1)
-        struct_keys = Map.keys(source.__struct__())
-
-        if Enum.member?(struct_keys, key) do
-          from(
-            [c] in query,
-            where: field(c, ^key) == ^value
-          )
-        else
-          query
+          is_map(params) and Map.has_key?(params, @key) ->
+            Map.get(params, @key)
+
+          is_map(params) and Map.has_key?(params, Atom.to_string(@key)) ->
+            Map.get(params, Atom.to_string(@key))
+
+          true ->
+            []
```

Comment:
I'm not sure I appreciate all of the use cases, but I wonder if it would make sense to complain a bit in this case and maybe also on [line 49](https://github.com/revelrylabs/ecto_filters/pull/1/files#diff-1a941b23f44ad9a5bd6c6774cc0ba8ffR49).

I'm just thinking... is somebody going to write filters and think they're working fine but in reality they're not doing anything b/c they're using the wrong syntax somehow?

Reply:
I wanted to accept params from conn so they're going to be maps with string keys and can be anything cause they come from the browser. That's why I put the apply inside the try on line 49. 

About the conditions in `get_filter_params`, we have to support maps with string keys, but I also want to make use of ecto_filters from inside the app and I either want atomized maps or keyword lists... and couldn't decide between the two.

Someone is definitely going to do that. but again I don't think I want to complain.

Reply:
I see... I get a weird feeling about the idea of passing unknown data into your DB queries (if a user knows what filters are on your schema, they could trigger ones you don't expect by crafting their payloads in a certain way), but probably not a big deal.

Reply:
It's a good feeling to have. I think we might want to have some "guidelines/notes" added somewhere and hope that `params` from a controller don't get passed directly into it without any thought.